Updated the C++ sample code for Textures and Mappings

jussiaaltonen opened the pull request.

dan-rigdon-bel commented:
@jussiaaltonen Sorry this took me so long to get to.  I've been sick the past week.

I read this over looking for grammar, clarity, etc.  I didn't run the sample.  My only feedback is that I think that the Overview section's

> **Texture Coordinates**: In Rhino these are 2d and 3d points that are saved in an ON_Mesh in the `m_T[]` or `m_TC[]` arrays.

bullet item should be updated to reflect surface parameters.  Said differently: if you refer to these in the sample, I think the names need to match here...if not, this bullet item needs to be updated to match the sample.

Otherwise the rest looks good to me.

jussiaaltonen commented:
@dan-rigdon-bel Thanks for your review.
I updated the texture coordinates bullet and added a new bullet for surface parameters (and increased the number of concepts accordingly).
Also added to the sample description that surface parameter mapping is getting applied.

Does that clarify the sample code?

dan-rigdon-bel commented:
Looks good to me @jussiaaltonen !
